modmesh/app/euler1d.py

```diff
@@ -442,288 +453,266 @@ def _update_single_figure(self):
-        the parameters they want to plot on the chart.
-
-        :param checked: button is checked or not
-        :return: nothing
+        :return: None
         """
-        checkbox = self.sender()
-        if self.checkbox_select_num == 3:
-            if checkbox.isChecked():
-                checkbox.toggle()
-            else:
-                self.checkbox_select_num -= 1
-                self.data_lines[checkbox.text()][1] = False
-        elif self.checkbox_select_num == 1:
-            if not checkbox.isChecked():
-                checkbox.toggle()
-            else:
-                self.checkbox_select_num += 1
-                self.data_lines[checkbox.text()][1] = True
-        else:
-            if checkbox.isChecked():
-                self.checkbox_select_num += 1
-                self.data_lines[checkbox.text()][1] = True
-            else:
-                self.checkbox_select_num -= 1
-                self.data_lines[checkbox.text()][1] = False
-
-        self._update_single_figure()
-
+        if self.max_steps and self.current_step > self.max_steps:
+            self.stop()
+            return
 
-class Controller:
-    def __init__(self, shocktube, max_steps, use_sub=None, profiling=False):
-        if None is shocktube.gamma:
-            raise ValueError("shocktube does not have constant built")
-        if None is shocktube.svr:
-            raise ValueError("shocktube does not have numerical solver built")
+        self.st.svr.march_alpha2(steps=steps)
+        self.current_step += steps
+        time_current = self.current_step * self.st.svr.time_increment
+        self.st.build_field(t=time_current)
+        cfl = self.st.svr.cfl
+        self.log(f"CFL: min {cfl.min()} max {cfl.max()}")
+        self.update_lines()
+        if self.profiling:
+            self.log(mm.time_registry.report())
 
-        super().__init__()
+    def step(self, steps=1):
+        """
+        Callback function of step button.
 
-        self.shocktube = shocktube
+        :return: None
+        """
+        self.march_alpha2(steps=steps)
 
-        self.max_steps = max_steps
-        self.current_step = 0
-        self.timer = None
-
-        self.use_sub = mm.Toggle.instance.get_value('apps.euler1d.use_sub',
-                                                    False)
-        if self.use_sub is None:
-            self.use_sub = mm.Toggle.instance.solid.use_pyside
-        self._main = QWidget()
-
-        if self.use_sub:
-            # FIXME: sub window causes missing QWindow with the following
-            # error:
-            # RuntimeError:
-            # Internal C++ object (PySide6.QtGui.QWindow) already deleted.
-            # It is probably because RMainWindow is not recognized by PySide6
-            # and matplotlib.  We may consider to use composite for QMainWindow
-            # instead of inheritance.
-            self._subwin = view.mgr.addSubWindow(self._main)
-            self._subwin.resize(1150, 750)
```

Comment:
Since the Euler1d app is constructed with dockable widgets and sub-windows, the option of using sub-windows is meaningless.